File: CW/Lista_1/Z4.scala
def sqrList(xs: List[Int]): List[Int] = {
    if(xs.isEmpty) {
        List()
    } else {
        xs.head*xs.head :: sqrList(xs.tail)
    }
}

@main def main() = {
    val list = List(1, 2, 3)
    val squareList = (list: List[Int]) => list.map(x => x*x)
    println(sqrList(list))
    println(squareList(list))
}

File: CW/Lista_1/Z4.ml
let rec sqrList (xs: int list): int list =
  if xs = [] then
    []
  else
    let head = List.hd xs in
    let tail = List.tl xs in
    head * head :: sqrList(tail)

let lst = [1;2;3];;
assert (sqrList lst = [1;4;9])